egs/aishell/s10/local/run_tdnn_1b.sh

```diff
@@ -0,0 +1,179 @@
+num_jobs_initial=2
+num_jobs_final=12
+minibatch_size=128
+frames_per_eg=150,110,90
+remove_egs=true
+common_egs_dir=
+xent_regularize=0.1
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  echo "This script is intended to be used with GPUs"
+  echo "but you have not compiled Kaldi with CUDA"
+  echo "If you want to use GPUs (and have them), go to src/,"
+  echo "and configure and make on a machine where "nvcc" is installed."
+  exit 1
+fi
+
+dir=${dir}${affix:+_$affix}
+train_set=train
+ali_dir=exp/tri5a_ali
+lat_dir=exp/tri5a_lats
+treedir=exp/chain_nnet3/tri5_tree
+lang=data/lang_chain_nnet3
+
+
+if [[ $stage -le 0 ]]; then
+  for datadir in train dev test; do
+    dst_dir=data/fbank_pitch/$datadir
+    if [[ ! -f $dst_dir/feats.scp ]]; then
+      utils/copy_data_dir.sh data/$datadir $dst_dir
+      echo "making fbank-pitch features for LF-MMI training"
+      steps/make_fbank_pitch.sh --cmd $train_cmd --nj $nj $dst_dir || exit 1
+      steps/compute_cmvn_stats.sh $dst_dir || exit 1
+      utils/fix_data_dir.sh $dst_dir
+    else
+      echo "$dst_dir/feats.scp already exists."
+      echo "kaldi pybind (local/run_chain.sh) LF-MMI may have generated it."
+      echo "skip $dst_dir"
+    fi
+  done
+fi
+
+if [[ $stage -le 1 ]]; then
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [[ $stage -le 2 ]]; then
+  # Build a tree using our new topology. This is the critically different
+  # step compared with other recipes.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --context-opts "--context-width=2 --central-position=1" \
+      --cmd $train_cmd 5000 data/train $lang $ali_dir $treedir
+fi
+
+if [[ $stage -le 3 ]]; then
+  echo "creating neural net configs using the xconfig parser"
+
+  num_targets=$(tree-info $treedir/tree | grep num-pdfs | awk '{print $2}')
+  learning_rate_factor=$(echo "print(0.5/$xent_regularize)" | python3)
+  feat_dim=$(feat-to-dim scp:data/fbank_pitch/train/feats.scp -)
+
+  mkdir -p $dir/configs
+  cat <<EOF > $dir/configs/network.xconfig
+  input dim=$feat_dim name=input
+
+  # please note that it is important to have input layer with the name=input
+  # as the layer immediately preceding the fixed-affine-layer to enable
+  # the use of short notation for the descriptor
+  fixed-affine-layer name=lda input=Append(-1,0,1) affine-transform-file=$dir/configs/lda.mat
```

Comment:
How much do we lose from removing i-vectors?  If you could make a comparison with run_tdnn_1a.sh via compare_wer.sh and put it in a comment at the top, that would be ideal.   (If there is no compare_wer.sh, please
see if someone over there can make one for this setup!).

Reply:
I did not use `ivector` since I have not figured it out how to integrate it into PyTorch.
I will try to add `ivector` and compare the results with/without using `ivector`.